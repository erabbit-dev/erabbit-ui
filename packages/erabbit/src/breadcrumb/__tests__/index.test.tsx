import { mount } from '@vue/test-utils'
import { Breadcrumb, BreadcrumbItem } from '..'

test('should render default Carousel', () => {
  const wrapper = mount(
    <Breadcrumb separator="&gt;">
      <BreadcrumbItem to="/">首页</BreadcrumbItem>
      <BreadcrumbItem to="/">活动管理</BreadcrumbItem>
      <BreadcrumbItem>活动编辑</BreadcrumbItem>
    </Breadcrumb>,
  )
  expect(wrapper.html()).toMatchSnapshot()
})

test('should add Transition component use animate', async () => {
  const wrapper = mount<any>({
    render() {
      return (
        <Breadcrumb separator="&gt;">
          <BreadcrumbItem to="/">首页</BreadcrumbItem>
          <BreadcrumbItem to="/" onClick={() => (this.show = !this.show)}>
            活动管理
          </BreadcrumbItem>
          {this.show && <BreadcrumbItem>活动编辑</BreadcrumbItem>}
        </Breadcrumb>
      )
    },
    data() {
      return {
        show: true,
      }
    },
  })

  await wrapper.find('.er-breadcrumb-item:nth-child(2)').trigger('click')

  expect(wrapper.findAll('.er-breadcrumb-item').length).toBe(2)

  await wrapper.find('.er-breadcrumb-item:nth-child(2)').trigger('click')

  expect(wrapper.findAll('.er-breadcrumb-item').length).toBe(3)
})
