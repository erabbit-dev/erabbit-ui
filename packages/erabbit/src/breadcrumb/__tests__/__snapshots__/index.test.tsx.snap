// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render default Breadcrumb 1`] = `
"<div class=\\"er-breadcrumb\\">
  <div class=\\"er-breadcrumb-item\\"><a href=\\"javascript:;\\">首页</a>
    <!---->
  </div>
  <div class=\\"er-breadcrumb-item\\"><a href=\\"javascript:;\\">活动管理</a>
    <!---->
  </div>
  <div class=\\"er-breadcrumb-item\\"><span>活动编辑</span>
    <!---->
  </div>
</div>"
`;
